### PR TITLE
feat: enable iframe use

### PR DESCRIPTION
**Pull request by livio-a**
Bildschirmfoto 2022-12-07 um 15 33 34

### CODE REVIEW

File: proto/zitadel/admin.proto [hunk 1]
```diff
@@ -2929,6 +2952,27 @@ message UpdateOIDCSettingsResponse {
...
+    zitadel.settings.v1.SecurityPolicy policy = 1;
+}
+
+message SetSecurityPolicyRequest{
+    message Disabled{}
+    oneof setting {
+        option (validate.required) = true;
+
+        Disabled disabled = 1;
+        zitadel.settings.v1.Origins enabled = 2;
```
Comment (hifabienne) [inline proto/zitadel/admin.proto:2968]: to make it more clear we could say enabled_origins? or do you think its enough only enabled?

File: internal/api/authz/instance.go [hunk 2]
```diff
@@ -66,6 +67,10 @@ func (i *instance) DefaultOrganisationID() string {
 	return i.orgID
 }
 
+func (i *instance) SecurityPolicyAllowedOrigins() []string {
+	return nil
```
Comment (hifabienne) [inline internal/api/authz/instance.go:71]: Not sure, but I did not find this method doing something somewhere? Shouldn't this return something?

**Review by hifabienne — COMMENTED** [review]
Please add some docs

File: internal/api/authz/instance.go (hunk 2 above)
Comment (livio-a) [reply internal/api/authz/instance.go:71]: this function is part of the authz.Instance interface and is used in the SecurityHeaders ServeHTTP function
the mock is currently not needed for that part, but must implement the whole interface

File: console/src/assets/i18n/de.json [hunk 3]
```diff
@@ -881,7 +881,8 @@
       "BRANDING": "Branding",
       "PRIVACYPOLICY": "Datenschutzrichtlinie",
       "OIDC": "OIDC Token Lifetime und Expiration",
-      "SECRETS": "Secret Erscheinungsbild"
+      "SECRETS": "Secret Erscheinungsbild",
```
Comment (hifabienne) [inline console/src/assets/i18n/de.json:884]: It this on purpose half in english?

File: docs/docs/guides/solution-scenarios/configurations.mdx [hunk 4]
```diff
@@ -74,3 +74,37 @@ Go to the "Advanced" section, per default login with email address should be all
...
+:::
+
+If your applications need to load ZITADEL inside an iframe, e.g. for a silent login or silent refresh, you can enable the use on an instance level.
+
+1. Navigate to the Instance Settings.
+2. Click on the Security Policy tab.
+3. Enable the "IFrame use" and add the host(s) you load the iframe from.
+
+You can add further hosts later on.
+[//]: # (TODO: add image)
```
Comment (hifabienne) [inline docs/docs/guides/solution-scenarios/configurations.mdx:102]: Image missing

Comment (livio-a) [reply docs/docs/guides/solution-scenarios/configurations.mdx]: added

File: console/src/assets/i18n/de.json (hunk 3 above)
Comment (livio-a) [reply console/src/assets/i18n/de.json:884]: i checked the current translation, we always have "secret" as in client secret, ...

**Review by hifabienne — APPROVED** [review]

**Pull request merged**